lib/gemonames/results.rb

```diff
@@ -0,0 +1,77 @@
+module Gemonames
+  SearchResult = Value.new(
+    :geoname_id,
+    :name,
+    :country_code,
+    :admin_id4,
+    :admin_id3,
+    :admin_id2,
+    :admin_id1,
+    :country_id,
+    :result,
+  ) do
+    alias_method :result?, :result
+    alias_method :present?, :result
+
+    def self.with_nil
```

Comment:
I wasn't sure about `with_nil` vs `with_nils`. Both sound off to be honest, but I like the symmetry with `with` which is what we use for non-null objects.